pcdsdevices/lens.py

```diff
@@ -286,31 +341,294 @@ def _make_safe(self):
             filt.insert()
             time.sleep(0.01)
         if filt.inserted:
-            print("REMINDER: Beam stop attenuator moved in!")
+            logger.info('Beam stop attenuator moved in!')
             safe = True
         else:
-            print("WARNING: Beam stop attenuator did not move in!")
+            logger.warning('Beam stop attenuator did not move in!')
             safe = False
         return safe
 
 
 class LensStack(LensStackBase):
+    """
+    Class for Be lens.
+
+    Parameters
+    ----------
+    x_prefix : str
+        The EPICS prefix that identifies the x motor.
+    y_prefix : str
+        The EPICS prefix that identifies the y motor.
+    z_prefix : str
+        The EPICS prefix that identifies the z motor.
+    lens_set : list, optional
+        List of lens sets. e.g. [numer1, lensthick1, number2, lensthick2...]
+    z_offset : number
+        Distance from sample to lens_z=0 in meters.
+    z_dir : number
+        1 or -1, represents beam direction wrt z direction.
+    E: number, optional
+        Beam energy
+    att_obj : attenuator object, optional
+    lcls_obj
+        Object that gets PVs from lcls (for energy)
+    mono_obj
+        Object that gets energy from monochromator
+    beamsize_unfocused : float, optional
+        Radial size of x-ray beam before focusing.
+    path : str
+        Path to the file that defines which lenses are being used.
+
+    Examples
+    --------
+
+    Before using the LclsStack class, provide the path of the be lens set file
+    to be used for the calculations.
+
+    If no lens sets are added in the file yet, use the
+    `be_lens_calcs.set_lens_set_to_file` function to set the lens sets:
+
+    >>> import pcdsdevices.lens as lens
+    >>> import pcdscalc.be_lens_calcs as be
+
+    >>> path = '../path/to/lens_set'
+
+    >>> sets_list = [[3, 0.0001, 1, 0.0002],
+                     [1, 0.0001, 1, 0.0003, 1, 0.0005],
+                     [2, 0.0001, 1, 0.0005]]
+    >>> be.set_lens_set_to_file(sets_list, path)
+
+    Create the LensStack() object by providing the x, y and z `prefixes`, as
+    well as all the other parameters:
+
+    >>> att = Attenuator('att', 4, name='att')
```

Comment:
When calling `move()` we need to provide an attenuator object, how do scientists go about? Is this somewhat they way we do it (of course providing a real prefix instead of a dummy string)

Reply:
This is a huge problem with the existing class- there isn't a reasonable way to connect these objects together. In the hutch python session there will be an existing attenuator object, my constructor in xpp's beamline file looks like:
```
with safe_load('xpp_lens'):
    from pcdsdevices.lens import LensStack
    from xpp.db import xpp_attenuator
    _folder = "/reg/g/pcds/pyps/apps/hutch-python/xpp/"
    xpp_lens = LensStack("XPP:SB2:MMS:13",  # X
                         "XPP:SB2:MMS:14",  # Y
                         "XPP:SB2:MMS:15",  # Z
                         name='xpp_lens',
                         z_offset=3.852,   # lifted from old
                         z_dir=-1,         # lifted from old
                         E=9.5,            # keV, not automatic
                         att_obj=xpp_attenuator,  # breaks happi
                         beamsize_unfocused=500e-6,  # from old
                         path=_folder + "lens/sets_Be")
```

Reply:
Ok i see, so scientists are able to just use the existing attenuator then - but there isn't much I can do here for now right? I'm just going to update the docstring

Reply:
Makes sense

Reply:
@ZLLentz can I assume that in the python session there will be an existing lcls object too?

Reply:
yes, there will be an existing lcls object